modules/apps/transfer/types/ack.go

```diff
@@ -0,0 +1,29 @@
+package types
+
+import (
+	"fmt"
+
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+)
+
+// NewForwardErrorAcknowledgement returns a new error acknowledgement with path forwarding information.
+func NewForwardErrorAcknowledgement(packet channeltypes.Packet, ack channeltypes.Acknowledgement) channeltypes.Acknowledgement {
+	ackErr := fmt.Sprintf("forward packet error: source: %s/%s destination: %s/%s %s",
```

Comment:
happy including both source and port, I think just the source makes sense as well (should match the hops specified on the original send)
```suggestion
	ackErr := fmt.Sprintf("forwarding packet failed on %s/%s: %s",
```

Reply:
Aye, indeed I suppose we could omit `transfer` (port id) but no harm including it. 

And yeah! You're right, I think just using source ends up matching the hops specified from the original send, and that's fair enough I suppose. Whoever constructs a msg with forwarding info never has to specify the destination port/channel, this is always worked out internally by ibc so this is fine imo 👍🏻 

I'll commit your suggestion and adjust the tests. Would you suggest adding the constructors to `forwarding.go` or happy to leave in `ack.go`? I'm fine with either